test/sql/function/list/lambdas/list_comprehension.test

```diff
@@ -0,0 +1,28 @@
+# name: test/sql/function/list/lambdas/list_comprehension.test
+# description: Test different examples that can be found when searching for python list comprehension examples
+# group: [lambdas]
+
+statement ok
+PRAGMA enable_verification
```

Comment:
Can we add tests with :
* Parameters inside lambda functions (see e.g. `test/sql/prepared`)
* Positional references inside lambda functions (e.g. `SELECT list_transform([1, 2, 3], x -> x + #1) FROM range(10)`
* Calling `list_filter` and `list_transform` with the wrong number of arguments (e.g. `list_transform([])`, `list_transform()`, ...)
* Calling `list_filter` and `list_transform` with the wrong argument types (e.g. `list_transform(1, 2)`, `list_transform(1, x -> x + 1)`, `list_transform([], 1)`
* Using lambda functions in functions that do not support it (e.g. `SELECT cos(x -> x + 1)`)